src/pages/PoolDetails/PoolDetailsStats.tsx

```diff
@@ -0,0 +1,229 @@
+  justify-content: space-between;
+
+  @media (max-width: ${BREAKPOINTS.lg - 1}px) {
+    flex-direction: column;
+  }
+`
+
+const PoolBalanceTokenNames = styled(Row)`
+  font-weight: 485;
+  font-size: 18px;
+  line-height: 24px;
+  width: max-content;
+
+  @media (max-width: ${BREAKPOINTS.lg - 1}px) {
+    font-size: 20px;
+    line-height: 28px;
+    width: 100%;
+  }
+`
+
+const leftBarChartStyles = css`
+  border-top-left-radius: 5px;
+  border-bottom-left-radius: 5px;
+  border-right: 1px solid ${({ theme }) => theme.surface2};
+`
+
+const rightBarChartStyles = css`
+  border-top-right-radius: 5px;
+  border-bottom-right-radius: 5px;
+  border-left: 1px solid ${({ theme }) => theme.surface2};
+`
+
+const BalanceChartSide = styled.div<{ percent: number; $color: string; isLeft: boolean }>`
+  height: 8px;
+  width: ${({ percent }) => percent * 100}%;
+  background: ${({ $color }) => $color};
+  ${({ isLeft }) => (isLeft ? leftBarChartStyles : rightBarChartStyles)}
+`
+
+interface PoolDetailsStatsProps {
+  poolData: PoolData
+  isReversed: boolean
+  chainId?: number
+}
+
+export function PoolDetailsStats({ poolData, isReversed, chainId }: PoolDetailsStatsProps) {
+  const isScreenSize = useScreenSize()
+  const screenIsNotLarge = isScreenSize['lg']
+  const { formatNumber } = useFormatter()
+
+  // We can't wrap this in a useMemo hook because useCurrency is also a hook
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const currencies = [
+    useCurrency(poolData?.token0?.id, chainId) ?? undefined,
+    useCurrency(poolData?.token1?.id, chainId) ?? undefined,
+  ]
+
+  // We can't wrap this in a useMemo hook because useColor is also a hook
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const color0 = useColor(currencies[0]?.wrapped)
+  const color1 = useColor(currencies[1]?.wrapped)
+
+  const [token0, token1] = useMemo(() => {
+    const fullWidth = poolData?.tvlToken0 / poolData?.token0Price + poolData?.tvlToken1
+    const token0FullData = {
+      ...poolData?.token0,
+      price: poolData?.token0Price,
+      tvl: poolData?.tvlToken0,
+      color: color0,
+      percent: poolData?.tvlToken0 / poolData?.token0Price / fullWidth,
+      currency: currencies[0],
+    }
+    const token1FullData = {
+      ...poolData?.token1,
+      price: poolData?.token1Price,
+      tvl: poolData?.tvlToken1,
+      color: color1,
+      percent: poolData?.tvlToken1 / fullWidth,
+      currency: currencies[1],
+    }
+    return isReversed ? [token1FullData, token0FullData] : [token0FullData, token1FullData]
+  }, [color0, color1, currencies, isReversed, poolData])
+
+  return (
+    <StatsWrapper>
+      <HeaderText>
+        <Trans>Stats</Trans>
+      </HeaderText>
+      <StatItemColumn>
+        <ThemedText.BodySecondary>
+          <Trans>Pool balances</Trans>
+        </ThemedText.BodySecondary>
+        <PoolBalanceSymbols>
+          <PoolBalanceTokenNames>
+            {!screenIsNotLarge && (
```

Comment:
could you use one of the higher order components for this ? https://github.com/Uniswap/interface/blob/main/src/theme/components/index.tsx#L457

Reply:
For these two logos a new one would need to be added that's displays only on Large and above

Reply:
This component has also only be used once, it seems like it has been nearly completely replaced with doing direct css styling or the useScreenSize hook